packages/yarnpkg-core/sources/tgzUtils.ts

```diff
@@ -39,11 +40,11 @@ export interface ExtractBufferOptions {
 
 let workerPool: WorkerPool<ConvertToZipPayload, PortablePath> | null;
 
-export async function convertToZip(tgz: Buffer, opts: ExtractBufferOptions) {
+export async function convertToZip(tgz: Buffer, limit: Limit, opts: ExtractBufferOptions) {
   const tmpFolder = await xfs.mktempPromise();
   const tmpFile = ppath.join(tmpFolder, `archive.zip`);
 
-  workerPool ||= new WorkerPool(getZipWorkerSource());
+  workerPool ||= new WorkerPool(getZipWorkerSource(), limit);
```

Comment:
Similar to the previous point, I'd prefer if we used an option bag rather than a free parameter.

Reply:
Added an optional option bag to `WorkerPool`.